### PR TITLE
Asmfixes

### LINKED ISSUE #4460: [Linux] Just a Black Screen
If you get a black screen on linux, please add a comment to this issue with the following information:
- Linux Distro
- Graphics Card
- Graphics Driver version
and please test if it happens with both the AppImage and with self-compiled version.

### PR DESCRIPTION
**Pull request by Nekotekina**
May be able to fix #4460

### CODE REVIEW

**Comment by apathyrecharge** [comment]
can confirm, "works on my machine ;)" as the old saying goes.

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 1]
```diff
@@ -1676,38 +1750,45 @@ s64 SPUThread::get_ch_value(u32 ch)
...
 		{
-			waiter.owner = this;
-			waiter.addr = raddr;
-			waiter.size = 128;
-			waiter.stamp = rtime;
-			waiter.data = rdata.data();
-			waiter.init();
+			if (mask1 != SPU_EVENT_LR)
+			{
+				fmt::throw_exception("Not supported: event mask 0x%x" HERE, mask1);
```
[the remark this answers is not in the document]
Comment (Nekotekina) [reply rpcs3/Emu/Cell/SPUThread.cpp]: This and other comments are inconsistent with the new code, TM is fixed though.

**Comment by Filoppi** [comment]
God of War Collection (BCES00791) doesn't start anymore after these 2 commits.

If you try to open the game, the game windows doesn't even open.
I don't have time to debug in VS now.
A clean log is attached
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/2010677/RPCS3.log)

**Comment by Xcedf** [comment]
MGS3, RDR, Killzone 2, no longer works after 367f0395, games stuck the emulator

**Comment by blottto** [comment]
Now TSX does not work in Haswell :( from rpcs3-v0.0.5-6750-367f0395 upwards
any solution ?

**Comment by rodjun** [comment]
@blottto There's no solution for a broken processor, TSX is broken on haswell and this is bound to happen when people force-enable it

**Pull request merged**